eslint/babel-eslint-parser/src/configuration.cts

```diff
@@ -13,9 +13,7 @@ export = function normalizeESLintConfig(options: any) {
   return {
     babelOptions: { cwd: process.cwd(), ...babelOptions },
     ecmaVersion: ecmaVersion === "latest" ? 1e8 : ecmaVersion,
-    // https://eslint.org/docs/latest/use/configure/language-options#specifying-javascript-options
-    // ESLint supports "commonjs" but Babel parser does not.
-    sourceType: sourceType === "commonjs" ? "script" : sourceType,
+    sourceType,
```

Comment:
`@babel/core` is a peer dependency, we should probably check if we are on a version that supports commonjs.

Reply:
OK, though it was the previous behavior before https://github.com/babel/babel/pull/17349.